File: practice_java_program/notpad/FirstandLastOccurance.java
public class FirstandLastOccurance{
	public static void main(String[] args){
		String input = "Hello World";
        	char target = 'o'; 
		int firstindex=-1;
		int lastindex=-1;
		// at first index
		for(int i=0; i<input.length(); i++){
			if(input.charAt(i)==target){
				firstindex=i;
				break;
			}
		}
		// at last index
		for(int i=input.length()-1; i>=0; i--){
			if(input.charAt(i)==target){
				lastindex=i;
				break;
			}
		}
		if(firstindex<0){
			System.out.println("Character "+target+"is not present in string");
		}
		else{
			System.out.println("first index for "+target+" = " +firstindex);
			System.out.println("last index for "+target+" = " +lastindex);
		}
	}
}

File: python_oriented_interview/notpad/FirstandLastOccurance.py
input_str = "Hello World"
target = 'o'
firstindex = -1
lastindex = -1

# first index
for i in range(len(input_str)):
    if input_str[i] == target:
        firstindex = i
        break

# last index
for i in range(len(input_str) - 1, -1, -1):
    if input_str[i] == target:
        lastindex = i
        break

if firstindex < 0:
    print(f"Character {target} is not present in string")
else:
    print(f"first index for {target} = {firstindex}")
    print(f"last index for {target} = {lastindex}")
